airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -72,6 +87,29 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
 
     final int version = Workflow.getVersion(VERSION_LABEL, Workflow.DEFAULT_VERSION, CURRENT_VERSION);
     final String taskQueue = Workflow.getInfo().getTaskQueue();
+
+    final int autoDetectSchemaVersion =
+        Workflow.getVersion(AUTO_DETECT_SCHEMA_TAG, Workflow.DEFAULT_VERSION, AUTO_DETECT_SCHEMA_VERSION);
+    if (autoDetectSchemaVersion >= AUTO_DETECT_SCHEMA_VERSION) {
+      try {
+        final UUID sourceId = configFetchActivity.getStandardSync(connectionId).getSourceId();
+        if (refreshSchemaActivity.shouldRefreshSchema(sourceId)) {
+          LOGGER.info("Refreshing source schema...");
+          refreshSchemaActivity.refreshSchema(sourceId, connectionId);
+        }
+        final Status status = configFetchActivity.getStandardSync(connectionId).getStatus();
+        if (Status.INACTIVE == status) {
+          LOGGER.info("Connection is disabled. Cancelling run.");
+          final StandardSyncOutput output =
+              new StandardSyncOutput()
+                  .withStandardSyncSummary(new StandardSyncSummary().withStatus(ReplicationStatus.CANCELLED).withTotalStats(new SyncStats()));
+          return output;
+        }
+      } catch (JsonValidationException | ConfigNotFoundException | IOException | ApiException e) {
+        LOGGER.error("An error occurred during schema refresh processing. Skipping schema refresh. ", e);
```

Comment:
Does it make sense to continue in this case? If we were unable to refresh the schema, I wonder if we're just risking getting into more errors.

Reply:
I think it does make the most sense to continue. Refreshing the schema every time isn't a hard requirement (we're not doing it now, for example), and I don't think we should block syncs just because we had an issue refreshing the schema.

Reply:
I wonder if we are taking the risk of running into not determinism errors here. Assuming it was a transient error, upon replay, we would get a different result here which I suspect may end up aborting the active sync.
If we want to be able to continue in this case, I think it would be safer to handle those errors within the activity and return a response with an error state so that temporal caches it. Unless temporal is caching errors, in which case, this not be an issue.

Reply:
To be tested but I think that temporal have the exception that is being thrown in its history and re-throw it.

Reply:
We have error handling within workflows already - I think we would've seen issues with this if this were a problem, or are you saying something different?

Reply:
Errors within the connection manaager workflow amke the workflow to continue as new so there is no risk of having an history replay issue. I think that we won't face an issue here but I think that it would worth testing it manually.

Reply:
We already have one within the SyncWorkflowImpl too on line 135 - i can test manually though

Reply:
How should I test this? I mean how can I 'replay' a run of a sync? Just trying syncing again?

Reply:
Looking at the temporal [retry policy doc](https://docs.temporal.io/retry-policies#default-behavior), I'd expect that temporal would catch the exception and retry the activity by default.

There is a way to set the retry policy for the activity (see [how-to-set-activity-retry-options-in-java](https://docs.temporal.io/java/how-to-set-activity-retry-options-in-java)).
```
ActivityOptions.newBuilder()
        .setRetryOptions(RetryOptions.newBuilder().setDoNotRetry().build())
```

One way to test might be to just throw the exceptions and see how the errors are currently handled.

Reply:
Temporal does retry the activity by default - I moved the exception handling into the activities instead